### ISSUE TITLE
`resample_filter` generates even-sized filters for even-sized upsample factors

### ISSUE THREAD

**Issue by ssfrr** [JuliaDSP/DSP.jl#138]
If `resample` is used to up-sample a signal by an even-number factor, the FIR filter generated with `resample_filter` ends up with an even number of taps, so after running the filter the original samples no longer line up.

I'm not 100% sure, but I think that the FIR filter used for upsampling should always have an odd size with value `1` in the center and `0`s every K samples.

```julia
using PlotlyJS, DSP

sig = rand(20)-0.5
K = 4
plot([
    stem(y=resample(sig, K//1)),
    stem(x=(0:length(sig)-1) * K + 1, y=sig)
    ])
```

![image](https://cloud.githubusercontent.com/assets/734386/20854842/33cdd2c0-b8c5-11e6-9d9d-e51ed557e402.png)

You can see how things line up correctly when `K==3`:

![image](https://cloud.githubusercontent.com/assets/734386/20854851/4b75655a-b8c5-11e6-8d69-953c92f4f460.png)

You can see the generated filter directly:

```julia
resamp_filter = resample_filter(4//1)
plot(stem(y=resamp_filter))
```

![image](https://cloud.githubusercontent.com/assets/734386/20854861/76095ff6-b8c5-11e6-84f6-768141822d6d.png)

**Comment by ssfrr** [comment]
After a little more digging I _think_ that what we should be doing is generating the odd-length lowpass FIR first and then zero-padding it to be a multiple of the resample numerator, rather than generating a filter that's bumped up to an even multiple.

Thoughts @JayKickliter?

If folks think that's reasonable I can do the PR, I'm just a little shaky on my polyphase theory so I don't want to mess anything up.

**Comment by mpghf** [comment]
I agree with ssfrr on this.  I came across the same issue.  For details on why you need an odd-length FIR filter to preserve the exact original data, see:

A Digital Signal Processing Approach to Interpolation, Ronald W. Schafer and Lawrence R. Rabiner, Proceedings of the IEEE, vol 61, pp 692-702, June 1973.  Page 160 in particular gives a good explanation.

**Comment by alecokas** [comment]
@ssfrr @mpghf  -  I too agree that the FIR filter should be odd. 

I have created a pull request which should address this issue as well as a time delay issue in the resample function -  #139 . I am not sure why the nightly build is failing but I do not think it is my fault as the 0.5 and 0.4 versions pass.

**Comment by alecokas** [comment]
After making the topographical changes suggested by @simonster at #139 all checks pass on the pull request

**Comment by simonster** [comment]
Should be fixed by #140
